src/panels/lovelace/special-rows/hui-select-row.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  html,
+  LitElement,
+  TemplateResult,
+  customElement,
+  property,
+  css,
+  CSSResult,
+} from "lit-element";
+
+import "@polymer/paper-checkbox/paper-checkbox";
+// Not duplicate import, it's for typing
+// tslint:disable-next-line
+import { PaperCheckboxElement } from "@polymer/paper-checkbox/paper-checkbox";
+import "../components/hui-generic-entity-row";
+import { EntityRow, EntityConfig } from "../entity-rows/types";
+import { HomeAssistant } from "../../../types";
+
+import "../../../components/ha-icon";
+import { fireEvent } from "../../../common/dom/fire_event";
+
+declare global {
+  // for fire event
+  interface HASSDomEvents {
+    "entity-selection-changed": EntitySelectionChangedEvent;
+  }
+}
+
+export interface EntitySelectionChangedEvent {
+  entity: string;
+  selected: boolean;
+}
+
+@customElement("hui-select-row")
+class HuiSelectRow extends LitElement implements EntityRow {
+  public hass?: HomeAssistant;
+
+  @property() private _config?: EntityConfig;
+
+  public setConfig(config: EntityConfig): void {
+    if (!config) {
+      throw new Error("Error in card configuration.");
+    }
+
+    this._config = config;
+  }
+
+  protected render(): TemplateResult | void {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+
+    return html`
+      <hui-generic-entity-row .hass="${this.hass}" .config="${this._config}">
+        <paper-checkbox
+          @click=${this._stopPropagation}
```

Comment:
We only have to stop propagation because generic entity row is going to open a more info dialog? I think that it's probably better in that case to just ditch hui-generic-entity-row. We should move to tables anyway 😛